org.hl7.fhir.r5/src/main/java/org/hl7/fhir/r5/comparison/CapabilityStatementComparer.java

```diff
@@ -0,0 +1,903 @@
+    r.getCells().add(style(gen.new Cell(null, null, right != null ? right.getValue().primitiveValue() : "", null, null), left != null ? left.getValue().primitiveValue() : null, right != null ? right.getValue().primitiveValue() : null, false));
+    r.getCells().add(gen.new Cell(null, null, "", null, null));
+    r.getCells().add(cellForMessages(gen, t.getMessages()));
+    return r;
+  }
+  
+  @SuppressWarnings("rawtypes")
+  private Row addPrimitiveTypeRow(HierarchicalTableGenerator gen, List<Row> rows, StructuralMatch<Element> t, CapabilityStatementComparison comparison) {
+    Row r = gen.new Row();
+    rows.add(r);
+    r.getCells().add(gen.new Cell(null, null, t.getName(), null, null));
+    PrimitiveType left = t.hasLeft() ? (PrimitiveType) t.getLeft() : null;
+    PrimitiveType right = t.hasRight() ? (PrimitiveType) t.getRight() : null;
+    r.getCells().add(style(gen.new Cell(null, null, left != null ? left.primitiveValue() : "", null, null), left != null ? left.primitiveValue() : null, right != null ? right.primitiveValue() : null, true));
+    r.getCells().add(gen.new Cell(null, null, "", null, null));
+    r.getCells().add(style(gen.new Cell(null, null, right != null ? right.primitiveValue() : "", null, null), left != null ? left.primitiveValue() : null, right != null ? right.primitiveValue() : null, false));
+    r.getCells().add(gen.new Cell(null, null, "", null, null));
+    r.getCells().add(cellForMessages(gen, t.getMessages()));
+    return r;
+  }
+  
+  private Cell style(Cell cell, String left, String right, boolean isLeft) {
+    if (left != null && right != null) {
+      if (!left.equals(right)) {
+        cell.setStyle("background-color: "+COLOR_DIFFERENT);
+      }
+    } else if (left != null) {
+      if (!isLeft) {        
+        cell.setStyle("background-color: "+COLOR_NO_CELL_RIGHT);
+      }
+    } else if (right != null) {        
+      if (isLeft) {        
+        cell.setStyle("background-color: "+COLOR_NO_CELL_LEFT);
+      }
+    }
+    return cell;
+  }
+
+//
+//  private Cell propertyCell(HierarchicalTableGenerator gen, ConceptDefinitionComponent cd, PropertyComponent p, boolean right, CapabilityStatementComparison comp) {
+//    ConceptPropertyComponent cp = getProp(cd, p, right, comp);
+//    if (cp == null) {
+//      return missingCell(gen, right ? COLOR_NO_CELL_RIGHT : COLOR_NO_CELL_LEFT);
+//    } else {
+//      return gen.new Cell(null, null, cp.getValue().toString(), null, null);
+//    }
+//  }
+//
+//  public ConceptPropertyComponent getProp(ConceptDefinitionComponent cd, PropertyComponent p, boolean right, CapabilityStatementComparison comp) {
+//    String c = p.getCode();
+//    if (right) {
+//      c = comp.getPropMap().get(c);
+//    }
+//    ConceptPropertyComponent cp = null;
+//    for (ConceptPropertyComponent t : cd.getProperty()) {
+//      if (t.getCode().equals(c)) {
+//        cp = t;
+//      }
+//    }
+//    return cp;
+//  }

```

Comment:
Do we need to leave the commented code in?

Reply:
oops removed. And there are unit tests